File: 5-31/q3.py
def part(M, left, right):

    out = left-1
    pivot = M[right]
    
    i = left
    while(i < right):
        if (M[i] > pivot):
            out+= 1

            temp = M[out]
            M[out] = M[i]
            M[i] = temp
        i +=1
    
    out += 1
    temp = M[right]
    M[right] = M[out]
    M[out] = temp

    return out

def sortC(M, low, high):

    if(low<high):
        
        pivot = part(M,low,high)

        sortC(M,low,pivot-1)
        sortC(M,pivot+1,high)
            



def func(array,k,n):
    sortC(array,0,n-1)

    # print(array)
    a,x=0,0
    b,y=0,0


    for i in range(n):
        if (y>=k and x>=k):
            break
        elif (y==k-1 and x==k):
            b+=array[i]
            b+=array[i+1]
            y+=1
            i+=1
        else:
            if (i%2 == 0):
                a+=array[i]
                x+=1
            else:
                b+=array[i]
                y+=1

    return max(a,b)



num = int(input())

for i in range(num):
    n,k = (input()).split(" ")
    n,k = int(n),int(k)
    array = input().split(" ")

    for i in range(n):
        array[i] = int(array[i])

    print(func(array,k,n))

File: 6-15(NCC Wildcard)/q3.java
import java.util.Scanner;

public class q3 {

    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        node[] p = new node[n];
        for (int i=0; i<3*n;i++){
            System.out.println(i);
            p[i].val = s.nextInt();
            p[i].index = i;
        }
    }

    static int func(node[] p, int n){

        int d=0;
        sortC(p,0,n-1);
        node maxIn = p[0];

        for (int i=0; i<n; i++){
            if (p[i].index>maxIn.index){
                maxIn = p[i];
            }
            d+=p[i].val;
        }

        node minIn = p[2*n];
        for (int i=2*n; i<3*n; i++){
            if (p[i].index<minIn.index){
                minIn = p[i];
            }
            d-=p[i].val;
        }

        if (maxIn.index<minIn.index){
            return d;
        }
        else{

            while(maxIn.index>=minIn.index) {

                if (d-maxIn.val+minIn.val <= d){

                    int i = 2*n-1;

                    while(true){
                        if (p[i].index>minIn.index){
                            d+=minIn.val;
                            minIn = p[i];
                            d-=minIn.val;
                            break;
                        }
                        i-=1;
                    }
                }

                else{

                    while(true){

                        int i = n;

                        if (p[i].index<maxIn.index){
                            d-=maxIn.val;
                            maxIn = p[i];
                            d+=maxIn.val;
                            break;
                        }
                        i+=1;
                    }

                }

            }
            return d;

        }
        
    }

    
    static int part(node[] M, int left, int right) {

        int out = left-1;
        int pivot = M[right].val;
        
        for(int i=left; i< right; i++) {
            if (M[i].val < pivot ) {
                out+= 1;

                int temp = M[out].val;
                M[out].val = M[i].val;
                M[i].val = temp;
            }
        
        }
        out += 1;
        int temp = M[right].val;
        M[right].val = M[out].val;
        M[out].val = temp;

        return out;
    }

    static void sortC(node[] M, int low, int high)
    {

        if(low<high) {
            
            int pivot = part(M,low,high);

            sortC(M,low,pivot-1);
            sortC(M,pivot+1,high);
            
        }
    }

}

class node{

    public int val;
    public int index;
}
